### PR TITLE
Support prequal edu modal

### PR DESCRIPTION
**Pull request by farhadn**
Added support for checking if prequal is supported. If so, when pressing the ALA button we show the prequal flow in SFSafariViewController

screen shot 2018-11-15 at 2 12 56 pm

### CODE REVIEW

File: AffirmSDK/AffirmAsLowAsButton.m [hunk 1]
```diff
@@ -39,18 +43,26 @@ - (void)configureWithAmount:(NSDecimalNumber *)amount
...
         [self layoutIfNeeded];
+        self.promoPrequalEnabled = promoPrequalEnabled;
         callback(success, error);
     }];
 }
 
 - (void)_showALAModal {
-    AffirmPromoModalViewController *promoModal = [AffirmPromoModalViewController promoModalControllerWithModalId:self.promoID amount:self.amount];
-    [self.presentingViewController presentViewController:promoModal animated:true completion:nil];
+    if (self.promoPrequalEnabled) {
```
Comment (rahulkgupta) [inline AffirmSDK/AffirmAsLowAsButton.m:57]: do we need this check? We can handle situations with the prequal when the promo does not have prequal enabled.

File: AffirmSDK/AffirmAsLowAsButton.m [hunk 2]
```diff
@@ -39,18 +43,26 @@ - (void)configureWithAmount:(NSDecimalNumber *)amount
...
+        NSString *prequalURL = [AffirmConfiguration sharedConfiguration].affirmPrequalURL;
+        NSString *url = [NSString stringWithFormat:@"%@?public_api_key=%@&unit_price=%@&promo_external_id=%@&isSDK=true&use_promo=True", prequalURL, [AffirmConfiguration sharedConfiguration].publicAPIKey, [AffirmNumberUtils decimalDollarsToIntegerCents:self.amount], self.promoID];
```
Comment (rahulkgupta) [inline AffirmSDK/AffirmAsLowAsButton.m:59]: @rohan @georgewitteman can you look at this from an API query params perspective?

I think it's not `unit_price` but `amount`, and everything but `public_api_key` is optional.

Comment (rahulkgupta) [reply AffirmSDK/AffirmAsLowAsButton.m:59]: Also what about `page_type`?

`page_type` is for differentiating between different promos.

**Review by rahulkgupta — COMMENTED** [review]
There's a notion of non-promoset promos that are still used by a few merchants. I don't think it will be a problem here, but just wanted to call it out.

**Review by hectorups — APPROVED** [review]

File: AffirmSDK/AffirmAsLowAsButton.m (hunk 1 above)
Comment (farhadn) [reply AffirmSDK/AffirmAsLowAsButton.m:57]: I decided to check this so we only use SFSafariViewController for prequal and leave the previous experience the same for everything else. So it's not needed to check it, but I think it's the right thing to do b/c it minimizes this change to just the case where it wasn't working before.

File: AffirmSDK/AffirmAsLowAsButton.m (hunk 2 above)
Comment (farhadn) [reply AffirmSDK/AffirmAsLowAsButton.m:59]: Any update on whether it's `amount` instead of `unit_price`? FWIW I tested with `unit_price` and it worked.

Comment (rohan) [reply AffirmSDK/AffirmAsLowAsButton.m:59]: It's `unit_price` for prequal.

File: AffirmSDK/AffirmAsLowAsButton.m (hunk 1 above)
Comment (rohan) [reply AffirmSDK/AffirmAsLowAsButton.m:57]: As of a few days ago, everything (except fast promos) are on the prequal app. The check you'll want is if the promo object returned by the backend has `config.promo_style == "fast"`.

File: AffirmSDK/AffirmAsLowAsButton.m (hunk 2 above)
Comment (georgewitteman) [reply AffirmSDK/AffirmAsLowAsButton.m:59]: @farhadn It should be `amount`, not `unit_price`. `unit_price` isn't anywhere in [the code](https://github.com/Affirm/all-the-things/blob/41fc91e3813f3b0684ca6cd4308848fe8039c5c5/promos/affirm/promos/flask.py#L217) so I'm not sure how that works for you.
```suggestion
        NSString *url = [NSString stringWithFormat:@"%@?public_api_key=%@&amount=%@&promo_external_id=%@&isSDK=true&use_promo=True", prequalURL, [AffirmConfiguration sharedConfiguration].publicAPIKey, [AffirmNumberUtils decimalDollarsToIntegerCents:self.amount], self.promoID];
```

I don't think I have enough context to say whether `page_type` is useful or not.

**Comment by rohan** [comment]
You'll probably want to cut over to the new promos endpoint, btw. This is using a very deprecated endpoint (/api/v2/promos), instead of the new one (/api/v2/promos/v2).

File: AffirmSDK/AffirmAsLowAsButton.m (hunk 2 above)
Comment (georgewitteman) [reply AffirmSDK/AffirmAsLowAsButton.m:59]: Sorry, didn't realize this is prequal. `unit_price` is correct.

**Pull request merged**